### PR TITLE
[feat](Nereids): use equal set to extend uniform slot and unique slots

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalPlan.java [hunk 1]
```diff
@@ -70,19 +71,25 @@ default FunctionalDependencies computeFuncDeps() {
...
+        List<Set<Slot>> uniqueSlots = fdBuilder.getAllUnique();
+        Set<Slot> uniformSlots = fdBuilder.getAllUniform();
         for (Slot slot : getOutput()) {
             Set<Slot> o = ImmutableSet.of(slot);
             // all slot dependents unique slot
-            for (Set<Slot> uniqueSlot : fdBuilder.getAllUnique()) {
+            for (Set<Slot> uniqueSlot : uniqueSlots) {
                 fdBuilder.addDeps(uniqueSlot, o);
             }
             // uniform slot dependents all unique slot
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalPlan.java:83]: this comment should be `uniform slot dependents all slot`?

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalPlan.java (hunk 1 above)
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalPlan.java:83]: Yeah, you are right. I'll fix it in the next pr

**Review by jackwener — APPROVED** [review]

**Comment by morrySnow** [comment]
run performance

**Pull request merged**